grpc/bcodes.go

```diff
@@ -85,6 +102,13 @@ func unwrapError(err error) error {
 		return core.LengthRequiredError(errBody)
 	case BadNonceError:
 		return core.BadNonceError(errBody)
+	case ProblemDetails:
+		pd := probs.ProblemDetails{}
+		jsonErr := json.Unmarshal([]byte(errBody), &pd)
+		if jsonErr != nil {
```

Comment:
You can inline the `Unmarshal` into this if. Also, this is not quite "fail open" as the comment says, it is more like "pass through an untyped error."